q2cli/builtin/tools.py

```diff
@@ -1237,3 +1237,138 @@ def supplement_replay(
+)
+@click.option(
+    '--text',
+    required=False,
+    help='Inline text for the Annotation.'
+         ' Mutually exclusive with `--file`.'
+)
+@click.option(
+    '--file',
+    'filepath',
+    required=False,
+    type=click.Path(exists=True, file_okay=True,
+                    dir_okay=False, readable=True),
+    help='Path to a text file whose contents will be added'
+         ' to your Annotation. Mutually exclusive with `--text`.'
+)
+@click.option(
+    '--output-path',
+    required=False,
+    type=click.Path(file_okay=True, dir_okay=False, writable=True),
+    help='Where to write the newly annotated result.'
+         '[default: overwrite input file]'
+)
+def annotation_create(input_path, annotation_type,
+                      name, text, filepath, output_path):
+    """Load a Result, add an Annotation and save."""
+    import qiime2.sdk
+    from qiime2.core.annotate import Note
+    from q2cli.core.config import CONFIG
+    from q2cli.util import get_plugin_manager
+
+    get_plugin_manager()
+    # enforce exactly one of `--text` or `--file`
+    if (text is None) == (filepath is None):
+        raise click.UsageError('Exactly one of `--text` or `--file`'
+                               ' must be provided.')
+
+    result = qiime2.sdk.Result.load(input_path)
+
+    # TODO: add support for additional annotation types post 7.0
+    if annotation_type == 'Note':
+        annotation = Note(name=name, text=text, filepath=filepath)
+    else:
+        raise click.BadParameter('Unsupported annotation type: '
+                                 f'{annotation_type}')
+
+    try:
+        result.add_annotation(annotation)
+    except Exception as e:
+        click.echo(CONFIG.cfg_style('error', str(e)), err=True)
+        raise click.Abort()
+
+    out_fp = output_path or input_path
+    result.save(out_fp)
+
+    click.echo(
+        CONFIG.cfg_style('success',
+                         f'Added {annotation_type} "{name}" to {out_fp}')
+    )
+
+
+@tools.command(
+    name='annotation-remove',
+    short_help='Remove an Annotation from a QIIME 2 Result.',
+    help='Remove an existing Annotation (by name) from'
+         ' an Artifact or Visualization.'
+)
+@click.option(
+    '--input-path',
+    required=True,
+    type=click.Path(exists=True, dir_okay=False, readable=True),
+    help='The `.qza` or `.qzv` from which to remove the Annotation.'
+)
+@click.option(
+    '--name',
+    required=True,
+    help='The name of the Annotation to remove.'
+)
+@click.option(
+    '--output-path',
+    required=False,
+    type=click.Path(dir_okay=False, writable=True),
+    help='Where to write the updated Result.'
+         '[default: overwrite input filepath]'
+)
+def annotation_remove(input_path, name, output_path):
+    """Load a Result, remove an Annotation by name and save."""
+    import qiime2.sdk
+    from q2cli.core.config import CONFIG
+    from q2cli.util import get_plugin_manager
+
+    get_plugin_manager()
```

Comment:
I don't think this is necessary here either. We don't care about the type or format or anything of the result we are annotating

Reply:
ok you're totally right - i gaslit myself on this bc i still had QIIMETEST set which is why none of the formats existed lol.